cpp/include/cudf/transform.hpp

```diff
@@ -25,6 +25,7 @@
 
 namespace CUDF_EXPORT cudf {
 
+/// @brief The type of transform to run
 enum class transform_type : int32_t {
   COLUMN      = 0,  ///< A transforms that outputs fixed-width types
   STRING_VIEW = 1   ///< A transform that outputs string-views that should be converted to strings
```

Comment:
Seems the output-type should determine this. I don't think this enum is necessary.

Reply:
Yes, I'll remove it. I was still trying to see what the API would look like in case we had different output types.
But since I'm settled on using `string_view` for the `string` UDF output anyway, I'll do just that.